### ISSUE TITLE
mesh manifest apply does not work with custom helm values

### ISSUE THREAD

**Issue by yacut** [istio/istio#17469]
**Bug description**

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
Istio install with custom helm values (grafana, prometheus etc.)

**Version (include the output of `istioctl version --remote` and `kubectl version`)**
```sh
✗ istioctl version --remote
client version: unknown
control plane version: release-1.3-20190927-10-16
✗ kubectl version                                                 
kubectlClient Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.0", GitCommit:"2bd9643cee5b3b3a5ecbd3af49d09018f0773c77", GitTreeState:"clean", BuildDate:"2019-09-19T13:57:45Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.6-eks-5047ed", GitCommit:"5047edce664593832e9b889e447ac75ab104f527", GitTreeState:"clean", BuildDate:"2019-08-21T22:32:40Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**
```yaml
# istio-control-plane.yaml
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
spec:
  values:
    global:
      logging:
        level: "default:info"
      logAsJson: false
      enableTracing: true
      mtls:
        enabled: true
    prometheus:
      enabled: true
    grafana:
      enabled: true
    tracing:
      enabled: true
      jaeger:
        memory:
          max_traces: 50000
        persist: false
    kiali:
      enabled: false
```

```sh
istioctl experimental manifest apply -f istio-control-plane.yaml
```

**Environment where bug was observed (cloud vendor, OS, etc)**
```sh
✗ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.14.6
BuildVersion:	18G103
```

**Comment by howardjohn** [comment]
what is the error?

**Comment by yacut** [comment]
@howardjohn I don't get any error, just no additional add-on (grafana, prometheus, kali etc) installed if I set the helm values for it. 

```sh
✗ kubectl --namespace istio-system get pod
NAME                                      READY   STATUS    RESTARTS   AGE
istio-citadel-84598d98b5-xn7gr            1/1     Running   0          34h
istio-egressgateway-867cdf4b5b-d2j6b      1/1     Running   0          34h
istio-galley-59b48b67f5-qrs44             2/2     Running   0          34h
istio-ingressgateway-77b9b654c9-5fj28     1/1     Running   0          10h
istio-ingressgateway-77b9b654c9-kfzmk     1/1     Running   0          10h
istio-pilot-7675b57bd5-s62p2              2/2     Running   0          34h
istio-policy-6669c776-g6fmz               2/2     Running   1          34h
istio-policy-6669c776-jlj5g               2/2     Running   0          12h
istio-sidecar-injector-557bd4c8c8-jtfbq   1/1     Running   0          34h
istio-telemetry-747d66bc5-jhgnj           2/2     Running   1          34h
istio-telemetry-747d66bc5-m2gpm           0/2     Pending   0          12h
```

I also noticed that the helm values for core components (e.g. istio-ingressgateway) works as expected, in this case, I've set replicas to 2, but grafana and prometheus are not installed
```yaml
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
spec:
  values:
    gateways:
      istio-ingressgateway:
        autoscaleEnabled: true
        autoscaleMin: 2
        autoscaleMax: 6
    prometheus:
      enabled: true
    grafana:
      enabled: true
```
The output of the `istioctl experimental manifest generate -f istio-control-plane.yaml ` command also does not contain grafana or prometheus deployment.

**Comment by yacut** [comment]
@howardjohn do you need more information?

**Comment by howardjohn** [comment]
Thanks for the info! This may be the same issue as https://github.com/istio/istio/issues/16970 which was recently fixed it looks like

**Comment by ostromart** [comment]
The fix just missed 1.3.2 but should be there in the next 1.3 release (and of course 1.4).

**Comment by yacut** [comment]
I can confirm that the version 1.3.3 fixes this issue 🎉

**Comment by yacut** [comment]
@howardjohn Unfortunately, I have noticed that not all custom values are applied.

For example:
```yaml
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
spec:
    kiali:
      enabled: true
      dashboard:
        auth:
          strategy: anonymous
        viewOnlyMode: false
```

This manifest should generate kiali config with dashboard setting, but it's not:
```sh
kubectl --namespace istio-system exec -it kiali-xxx-xxx -- cat /kiali-configuration/config.yaml
istio_namespace: istio-system
server:
  port: 20001
  web_root: /kiali
external_services:
  istio:
    url_service_version: http://istio-pilot.istio-system:8080/version
  jaeger:
    url:
  grafana:
    url:
```
Is it a bug or am I doing something wrong?

**Comment by howardjohn** [comment]
@richardwxn ^

**Comment by richardwxn** [comment]
@yacut 
The reason is that some of the values schema in operator is outdated, but you can always add --force flag to the manifest apply command bypass the validation and just pass through. I think that 's available for 1.4

Also you may need to make sure you put the kiali part under the values, the CR yaml you pasted most recently was not correct.

I can get below configmap using the values your provided, the auth strategy is being set correctly.
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: kiali
  namespace: istio-system
  labels:
    app: kiali
    release: istio
data:
  config.yaml: |
    istio_component_namespaces:
      grafana: istio-system
      tracing: istio-system
      pilot: istio-system
      prometheus: istio-system
    istio_namespace: istio-system
    auth:
      strategy: anonymous
    server:
      port: 20001
      web_root: /kiali
    external_services:
      istio:
        url_service_version: http://istio-pilot.istio-system:8080/version
      jaeger:
        url: 
      grafana:
        url:
```

**Comment by yacut** [comment]
@richardwxn Thank you for your answer. Then I'll wait until version 1.4 and test it again.
